test/testdata/rewriter/minitest_tables.rb

```diff
@@ -0,0 +1,53 @@
+# typed: true
+
+class A
+  def foo; end
+end
+class B < A
+end
+
+class MyTest
+  def outside_method
+  end
+
+  def self.test_each(arg, &blk); end
+  def self.it(name, &blk); end
+
+  test_each [A.new, B.new] do |value|
+    it "works with instance methods" do
+      puts value.foo
+      outside_method
+    end
+  end
+
+  CONST_LIST = [A.new, B.new]
+  test_each CONST_LIST do |value|
+    it "succeeds with a constant list" do
+      puts value.foo
+    end
+  end
+
+  local = [A.new, B.new]
+  test_each local do |value| # error: `test_each` can only be used with constants or array literals
```

Comment:
I anticipate that this will likely be a super frustrating limitation. Why is having a constant any better? A constant literal would just look like untyped.